src/api/users/src/routes/users.js

```diff
@@ -28,27 +28,35 @@ router.get('/:id', validateId(), async (req, res, next) => {
 });
 
 // get all users
-// rejected if the user collection is empty, otherwise users returned
-router.get('/', async (req, res, next) => {
-  try {
-    const usersRef = db.collection('users');
-    const snapshot = await usersRef.get();
+// rejected if the users collection is empty
+// otherwise n (specified via params) users are returned
+router.get('/', validatePagingParams(), async (req, res, next) => {
+  // if req.query.per_page is < 1 or unparsable, set to 100, otherwise set to the min of either req.query.page or 100
+  const parsePerPage = req.query.per_page < 1 ? 20 : Math.min(parseInt(req.query.per_page, 10), 20);
```

Comment:
Why are you assuming `req.query.per_page` is a number at the start of this (comparing to 1), but then you parse it as if it's a string later on?

Reply:
I added some default values to celebrate and removed all the validation logic we had in `users.js`, I figured since celebrate does the heavy lifting for us there's no reason to ensure that a number is an int, has a range, etc, since celebrate does all this for us and more.

What do you think?